### ISSUE TITLE
Error 'Er kan per flow maar 1 trigger gezet worden'

### ISSUE THREAD

**Issue by Fire69** [athombv/homey#1097]
**What is your Homey version (Settings → System)?**

1.0.1

**What did you try to achieve?**

Lots of times when I create or edit a flow and try to save it, I receive the error 'Er kan per flow maar 1 trigger gezet worden' even though 1 only have 1 card in the 'Als' column.
When I refresh the page - which causes me to lose my entire flow when creating a new one or loose my changes when editing - and start over, I am able to save.

Here is a log: CDC0098AB8
I edited a flow, tried to save it, got the error, refreshed, edited again and was able to save it.
Hopefully this is visible in the log.

**What did you expect as result?**

Saving the correctly the first time without having to start again.

**Any other remarks**

...

**Comment by WeeJeWel** [comment]
When this happens, could you open the Chrome Console and paste this line:

```
$('.cards[data-type="trigger"] > .item .cards-type.visible .card[data-order="0"]')
```

And make a screenshot of the output?

**Comment by Fire69** [comment]
Is this enough detail?
![image](https://cloud.githubusercontent.com/assets/1757287/20604907/210fedb0-b269-11e6-9965-0b7469f47c49.png)

If I understand, there are 2 trigger cards?
This is the flow:
![image](https://cloud.githubusercontent.com/assets/1757287/20605034/9a2158ec-b269-11e6-9e8e-f1bed6719d6f.png)

**Comment by WeeJeWel** [comment]
Whoo that's weird! Which browser are you using?

**Comment by Fire69** [comment]
Plain ol' Chrome :)

**Comment by Fire69** [comment]
Well, it happened again...

I edit a flow that turns off some lights when a variable changes and save it, no problem.
I edit another flow that turns the same lights on when the variable changes.  I get the error.
I try saving again, it gets saved. I F5 the page and all the cards from my previous are in this flow!!

![image](https://cloud.githubusercontent.com/assets/1757287/20862241/e4133c98-b9a5-11e6-9dca-9a6271879041.png)

Is there anything else logged that can be checked?
I'll try editing some flows in the desktop app to see what happens there.

**Comment by caseda** [comment]
i have had the same now twice in a row, and it happend on my laptop, didn't had it before on my desktop (both windows 10) so it might be a touchpad/touchscreen thing not activating the clear card memory properly.
this is the result, slightly unpacked:
```
0
:
div.card
accessKey
:
""
align
:
""
assignedSlot
:
null
attributes
:
NamedNodeMap
baseURI
:
"http://192.168.10.200/manager/flow/#067773a8-1eb4-4a21-919a-3c3b96ab00e9"
childElementCount
:
3
childNodes
:
NodeList[7]
children
:
HTMLCollection[3]
classList
:
DOMTokenList[1]
className
:
"card"
clientHeight
:
75
clientLeft
:
0
clientTop
:
0
clientWidth
:
250
contentEditable
:
"inherit"
dataset
:
DOMStringMap
dir
:
""
draggable
:
false
firstChild
:
text
firstElementChild
:
div.icon.bg.ui-draggable-handle
hidden
:
false
id
:
""
innerHTML
:
"↵↵								↵↵								↵↵								<div class="icon bg ui-draggable-handle" style="-webkit-mask-image: url( /app/com.fibaro/drivers/FGMS-001/assets/icon.svg )"></div>↵↵								↵								<div class="title">Slaapkamer</div>↵								↵↵								<span class="content">↵									<span class="title">↵↵										↵↵										<span title="De bewegingsmelder gaat uit">De bewegingsmelder gaat uit</span>↵									</span>↵									<span class="args">↵										↵									</span>↵↵								</span>↵↵								↵↵								↵↵								↵							"
innerText
:
"Slaapkamer↵De bewegingsmelder gaat uit↵"
isConnected
:
true
isContentEditable
:
false
lang
:
""
lastChild
:
text
lastElementChild
:
span.content
localName
:
"div"
namespaceURI
:
"http://www.w3.org/1999/xhtml"
nextElementSibling
:
null
nextSibling
:
text
nodeName
:
"DIV"
nodeType
:
1
nodeValue
:
null
offsetHeight
:
75
offsetLeft
:
0
offsetParent
:
div.cards-type.visible
offsetTop
:
0
offsetWidth
:
250
onabort
:
null
onbeforecopy
:
null
onbeforecut
:
null
onbeforepaste
:
null
onblur
:
null
oncancel
:
null
oncanplay
:
null
oncanplaythrough
:
null
onchange
:
null
onclick
:
null
onclose
:
null
oncontextmenu
:
null
oncopy
:
null
oncuechange
:
null
oncut
:
null
ondblclick
:
null
ondrag
:
null
ondragend
:
null
ondragenter
:
null
ondragleave
:
null
ondragover
:
null
ondragstart
:
null
ondrop
:
null
ondurationchange
:
null
onemptied
:
null
onended
:
null
onerror
:
null
onfocus
:
null
oninput
:
null
oninvalid
:
null
onkeydown
:
null
onkeypress
:
null
onkeyup
:
null
onload
:
null
onloadeddata
:
null
onloadedmetadata
:
null
onloadstart
:
null
onmousedown
:
null
onmouseenter
:
null
onmouseleave
:
null
onmousemove
:
null
onmouseout
:
null
onmouseover
:
null
onmouseup
:
null
onmousewheel
:
null
onpaste
:
null
onpause
:
null
onplay
:
null
onplaying
:
null
onprogress
:
null
onratechange
:
null
onreset
:
null
onresize
:
null
onscroll
:
null
onsearch
:
null
onseeked
:
null
onseeking
:
null
onselect
:
null
onselectstart
:
null
onshow
:
null
onstalled
:
null
onsubmit
:
null
onsuspend
:
null
ontimeupdate
:
null
ontoggle
:
null
ontouchcancel
:
null
ontouchend
:
null
ontouchmove
:
null
ontouchstart
:
null
onvolumechange
:
null
onwaiting
:
null
onwebkitfullscreenchange
:
null
onwebkitfullscreenerror
:
null
onwheel
:
null
outerHTML
:
"<div class="card" data-uri="homey:device:0f655513-d643-4059-a1d6-21f53af45303" data-type="trigger" data-id="alarm_motion_false" data-order="0" style="z-index: 7; transform: scale(1);">↵↵								↵↵								↵↵								<div class="icon bg ui-draggable-handle" style="-webkit-mask-image: url( /app/com.fibaro/drivers/FGMS-001/assets/icon.svg )"></div>↵↵								↵								<div class="title">Slaapkamer</div>↵								↵↵								<span class="content">↵									<span class="title">↵↵										↵↵										<span title="De bewegingsmelder gaat uit">De bewegingsmelder gaat uit</span>↵									</span>↵									<span class="args">↵										↵									</span>↵↵								</span>↵↵								↵↵								↵↵								↵							</div>"
outerText
:
"Slaapkamer↵De bewegingsmelder gaat uit↵"
ownerDocument
:
document
parentElement
:
div.cards-type.visible
parentNode
:
div.cards-type.visible
prefix
:
null
previousElementSibling
:
div.card
previousSibling
:
div.card
scrollHeight
:
75
scrollLeft
:
0
scrollTop
:
0
scrollWidth
:
250
shadowRoot
:
(...)
slot
:
""
spellcheck
:
true
style
:
CSSStyleDeclaration
tabIndex
:
-1
tagName
:
"DIV"
textContent
:
"↵↵								↵↵								↵↵								↵↵								↵								Slaapkamer↵								↵↵								↵									↵↵										↵↵										De bewegingsmelder gaat uit↵									↵									↵										↵									↵↵								↵↵								↵↵								↵↵								↵							"
title
:
""
translate
:
true
webkitdropzone
:
""
__proto__
:
HTMLDivElement
1
:
div.card
accessKey
:
""
align
:
""
assignedSlot
:
null
attributes
:
NamedNodeMap
baseURI
:
"http://192.168.10.200/manager/flow/#067773a8-1eb4-4a21-919a-3c3b96ab00e9"
childElementCount
:
3
childNodes
:
NodeList[7]
children
:
HTMLCollection[3]
classList
:
DOMTokenList[1]
className
:
"card"
clientHeight
:
75
clientLeft
:
0
clientTop
:
0
clientWidth
:
250
contentEditable
:
"inherit"
dataset
:
DOMStringMap
dir
:
""
draggable
:
false
firstChild
:
text
firstElementChild
:
div.icon.bg.ui-draggable-handle
hidden
:
false
id
:
""
innerHTML
:
"↵↵								↵↵								↵↵								<div class="icon bg ui-draggable-handle" style="-webkit-mask-image: url( /app/com.fibaro/drivers/FGPB-101/assets/icon.svg )"></div>↵↵								↵								<div class="title">Slaapkamer</div>↵								↵↵								<span class="content">↵									<span class="title">↵↵										↵↵										<span title="Push Button">Push Button</span>↵									</span>↵									<span class="args">↵										↵										<div class="arg flexible" data-type="dropdown" data-name="scene">↵	<select class="input flexible">↵		↵		<option value="0">1x ingedrukt</option>↵		↵		<option value="3">2x ingedrukt</option>↵		↵		<option value="4">3x ingedrukt</option>↵		↵		<option value="5">4x ingedrukt</option>↵		↵		<option value="6">5x ingedrukt</option>↵		↵		<option value="2">Lang Drukken (Blijft Activeren)</option>↵		↵		<option value="1">Lang Drukken Gestopt</option>↵		↵	</select>↵</div>↵										↵									</span>↵↵								</span>↵↵								↵↵								↵↵								↵							"
innerText
:
"Slaapkamer↵Push Button↵↵"
isConnected
:
true
isContentEditable
:
false
lang
:
""
lastChild
:
text
lastElementChild
:
span.content
localName
:
"div"
namespaceURI
:
"http://www.w3.org/1999/xhtml"
nextElementSibling
:
null
nextSibling
:
text
nodeName
:
"DIV"
nodeType
:
1
nodeValue
:
null
offsetHeight
:
75
offsetLeft
:
0
offsetParent
:
div.cards-type.visible
offsetTop
:
0
offsetWidth
:
250
onabort
:
null
onbeforecopy
:
null
onbeforecut
:
null
onbeforepaste
:
null
onblur
:
null
oncancel
:
null
oncanplay
:
null
oncanplaythrough
:
null
onchange
:
null
onclick
:
null
onclose
:
null
oncontextmenu
:
null
oncopy
:
null
oncuechange
:
null
oncut
:
null
ondblclick
:
null
ondrag
:
null
ondragend
:
null
ondragenter
:
null
ondragleave
:
null
ondragover
:
null
ondragstart
:
null
ondrop
:
null
ondurationchange
:
null
onemptied
:
null
onended
:
null
onerror
:
null
onfocus
:
null
oninput
:
null
oninvalid
:
null
onkeydown
:
null
onkeypress
:
null
onkeyup
:
null
onload
:
null
onloadeddata
:
null
onloadedmetadata
:
null
onloadstart
:
null
onmousedown
:
null
onmouseenter
:
null
onmouseleave
:
null
onmousemove
:
null
onmouseout
:
null
onmouseover
:
null
onmouseup
:
null
onmousewheel
:
null
onpaste
:
null
onpause
:
null
onplay
:
null
onplaying
:
null
onprogress
:
null
onratechange
:
null
onreset
:
null
onresize
:
null
onscroll
:
null
onsearch
:
null
onseeked
:
null
onseeking
:
null
onselect
:
null
onselectstart
:
null
onshow
:
null
onstalled
:
null
onsubmit
:
null
onsuspend
:
null
ontimeupdate
:
null
ontoggle
:
null
ontouchcancel
:
null
ontouchend
:
null
ontouchmove
:
null
ontouchstart
:
null
onvolumechange
:
null
onwaiting
:
null
onwebkitfullscreenchange
:
null
onwebkitfullscreenerror
:
null
onwheel
:
null
outerHTML
:
"<div class="card" data-uri="homey:device:848c5978-7faa-4677-a119-34fc4cba1a44" data-type="trigger" data-id="FGPB-101" data-order="0" style="↵									z-index: 1;↵									transform: scale( 1 );↵								">↵↵								↵↵								↵↵								<div class="icon bg ui-draggable-handle" style="-webkit-mask-image: url( /app/com.fibaro/drivers/FGPB-101/assets/icon.svg )"></div>↵↵								↵								<div class="title">Slaapkamer</div>↵								↵↵								<span class="content">↵									<span class="title">↵↵										↵↵										<span title="Push Button">Push Button</span>↵									</span>↵									<span class="args">↵										↵										<div class="arg flexible" data-type="dropdown" data-name="scene">↵	<select class="input flexible">↵		↵		<option value="0">1x ingedrukt</option>↵		↵		<option value="3">2x ingedrukt</option>↵		↵		<option value="4">3x ingedrukt</option>↵		↵		<option value="5">4x ingedrukt</option>↵		↵		<option value="6">5x ingedrukt</option>↵		↵		<option value="2">Lang Drukken (Blijft Activeren)</option>↵		↵		<option value="1">Lang Drukken Gestopt</option>↵		↵	</select>↵</div>↵										↵									</span>↵↵								</span>↵↵								↵↵								↵↵								↵							</div>"
outerText
:
"Slaapkamer↵Push Button↵↵"
ownerDocument
:
document
parentElement
:
div.cards-type.visible
parentNode
:
div.cards-type.visible
prefix
:
null
previousElementSibling
:
null
previousSibling
:
text
scrollHeight
:
75
scrollLeft
:
0
scrollTop
:
0
scrollWidth
:
250
shadowRoot
:
(...)
slot
:
""
spellcheck
:
true
style
:
CSSStyleDeclaration
tabIndex
:
-1
tagName
:
"DIV"
textContent
:
"↵↵								↵↵								↵↵								↵↵								↵								Slaapkamer↵								↵↵								↵									↵↵										↵↵										Push Button↵									↵									↵										↵										↵	↵		↵		1x ingedrukt↵		↵		2x ingedrukt↵		↵		3x ingedrukt↵		↵		4x ingedrukt↵		↵		5x ingedrukt↵		↵		Lang Drukken (Blijft Activeren)↵		↵		Lang Drukken Gestopt↵		↵	↵↵										↵									↵↵								↵↵								↵↵								↵↵								↵							"
title
:
""
translate
:
true
webkitdropzone
:
""
__proto__
:
HTMLDivElement
context
:
document
URL
:
"http://192.168.10.200/manager/flow/#067773a8-1eb4-4a21-919a-3c3b96ab00e9"
activeElement
:
body.manager.editing
alinkColor
:
""
all
:
HTMLAllCollection[19413]
anchors
:
HTMLCollection[0]
applets
:
HTMLCollection[0]
baseURI
:
"http://192.168.10.200/manager/flow/#067773a8-1eb4-4a21-919a-3c3b96ab00e9"
bgColor
:
""
body
:
body.manager.editing
characterSet
:
"UTF-8"
charset
:
"UTF-8"
childElementCount
:
1
childNodes
:
NodeList[2]
children
:
HTMLCollection[1]
compatMode
:
"CSS1Compat"
contentType
:
"text/html"
cookie
:
"i18next=nl; homey_cloud=1; homey_id=56b880c2161e20011a1b2806; homey_language=nl; homey_name=Homey; homey_version=1.0.4"
currentScript
:
null
defaultView
:
Window
designMode
:
"off"
dir
:
""
doctype
:
<!DOCTYPE html>
documentElement
:
html
documentURI
:
"http://192.168.10.200/manager/flow/#067773a8-1eb4-4a21-919a-3c3b96ab00e9"
domain
:
"192.168.10.200"
embeds
:
HTMLCollection[0]
fgColor
:
""
firstChild
:
<!DOCTYPE html>
firstElementChild
:
html
fonts
:
FontFaceSet
forms
:
HTMLCollection[0]
head
:
head
hidden
:
false
images
:
HTMLCollection[85]
implementation
:
DOMImplementation
inputEncoding
:
"UTF-8"
isConnected
:
true
jQuery213051856779307473631
:
4
lastChild
:
html
lastElementChild
:
html
lastModified
:
"11/18/2016 13:12:32"
linkColor
:
""
links
:
HTMLCollection[7]
location
:
Location
nextSibling
:
null
nodeName
:
"#document"
nodeType
:
9
nodeValue
:
null
onabort
:
null
onbeforecopy
:
null
onbeforecut
:
null
onbeforepaste
:
null
onblur
:
null
oncancel
:
null
oncanplay
:
null
oncanplaythrough
:
null
onchange
:
null
onclick
:
null
onclose
:
null
oncontextmenu
:
null
oncopy
:
null
oncuechange
:
null
oncut
:
null
ondblclick
:
null
ondrag
:
null
ondragend
:
null
ondragenter
:
null
ondragleave
:
null
ondragover
:
null
ondragstart
:
null
ondrop
:
null
ondurationchange
:
null
onemptied
:
null
onended
:
null
onerror
:
null
onfocus
:
null
oninput
:
null
oninvalid
:
null
onkeydown
:
null
onkeypress
:
null
onkeyup
:
null
onload
:
null
onloadeddata
:
null
onloadedmetadata
:
null
onloadstart
:
null
onmousedown
:
null
onmouseenter
:
null
onmouseleave
:
null
onmousemove
:
null
onmouseout
:
null
onmouseover
:
null
onmouseup
:
null
onmousewheel
:
null
onpaste
:
null
onpause
:
null
onplay
:
null
onplaying
:
null
onpointerlockchange
:
null
onpointerlockerror
:
null
onprogress
:
null
onratechange
:
null
onreadystatechange
:
null
onreset
:
null
onresize
:
null
onscroll
:
null
onsearch
:
null
onseeked
:
null
onseeking
:
null
onselect
:
null
onselectionchange
:
null
onselectstart
:
null
onshow
:
null
onstalled
:
null
onsubmit
:
null
onsuspend
:
null
ontimeupdate
:
null
ontoggle
:
null
ontouchcancel
:
null
ontouchend
:
null
ontouchmove
:
null
ontouchstart
:
null
onvolumechange
:
null
onwaiting
:
null
onwebkitfullscreenchange
:
null
onwebkitfullscreenerror
:
null
onwheel
:
null
origin
:
"http://192.168.10.200"
ownerDocument
:
null
parentElement
:
null
parentNode
:
null
plugins
:
HTMLCollection[0]
pointerLockElement
:
null
preferredStylesheetSet
:
null
previousSibling
:
null
readyState
:
"complete"
referrer
:
""
rootElement
:
null
scripts
:
HTMLCollection[35]
scrollingElement
:
body.manager.editing
selectedStylesheetSet
:
null
styleSheets
:
StyleSheetList
textContent
:
null
title
:
"Flow - Homey"
user-avatar
:
undefined
visibilityState
:
"visible"
vlinkColor
:
""
webkitCurrentFullScreenElement
:
null
webkitFullscreenElement
:
null
webkitFullscreenEnabled
:
true
webkitHidden
:
false
webkitIsFullScreen
:
false
webkitVisibilityState
:
"visible"
xmlEncoding
:
null
xmlStandalone
:
false
xmlVersion
:
null
__proto__
:
HTMLDocument
length
:
2
prevObject
:
Z.fn.init[1]
0
:
document
context
:
document
length
:
1
__proto__
:
Object[0]
selector
:
".cards[data-type="trigger"] > .item .cards-type.visible .card[data-order="0"]"
__proto__
:
Object[0]
```

**Comment by Fire69** [comment]
I had it yesterday again also, but then I switched to the Homey Desktop app and had no problem whatsoever, so I'm guessing it's a weird caching problem in Chrome.

**Comment by Fire69** [comment]
Don't know if you are still 'waiting for user feedback'?
On Slack, @msmits reported he had had the same problem.

**Comment by Fire69** [comment]
I'm having the problem now in the Homey Desktop app!

![image](https://cloud.githubusercontent.com/assets/1757287/21518830/42bc9482-cce9-11e6-92c7-196ea7b21768.png)

When I reload (Ctrl-R), it stays the same, so it seems like it's saved like this.
But when I use @irritanterik's Flow Viewer (https://github.com/irritanterik/homey-realtime-logpage), it only shows the cards once.
![image](https://cloud.githubusercontent.com/assets/1757287/21518897/b6fa7a6c-cce9-11e6-8773-38852aad9fce.png)

Hope this helps...

**Comment by irritanterik** [comment]
Or just do a GET on `http://<localip>/api/manager/flow/:flowid:` directly

**Comment by Fire69** [comment]
How do I do the GET?

Can't really test it right now, the problem is that when I go to another flow the problem disappears...
Will try to reproduce...

**Comment by WeeJeWel** [comment]
I think this is fixed in v1.3.0, if not, please report back then.
